packages/app-accounts/src/DownloadButton.tsx

```diff
@@ -0,0 +1,213 @@
+import { KeyringPair$Json } from '@polkadot/util-keyring/types';
+
+import React from 'react';
+import translate from './translate';
+import { Trans } from 'react-i18next';
+import FileSaver from 'file-saver';
+import keyring from '@polkadot/ui-keyring/index';
+import isUndefined from '@polkadot/util/is/undefined';
+
+import AddressMini from '@polkadot/ui-app/AddressMini';
+import Button from '@polkadot/ui-app/Button';
+import Modal from '@polkadot/ui-app/Modal';
+import Unlock from '@polkadot/ui-signer/Unlock';
+
+type State = {
+  address: string,
+  password: string,
+  isPasswordModalOpen: boolean,
+  error?: InputErrorMessage
+};
+
+type Props = I18nProps & BareProps & {
+  icon?: string,
+  isCircular?: boolean,
+  isPrimary?: boolean,
+  size?: Button$Sizes,
+  address: string
+};
+
+class DownloadButton extends React.PureComponent<Props, State> {
+  state: State;
+
+  constructor (props: Props) {
+    super(props);
+
+    this.state = this.emptyState();
+  }
+
+  handleDownloadAccount = (): void => {
+    const { t } = this.props;
+    const { address, password } = this.state;
+
+    if (!address) {
+      return;
+    }
+
+    // Reset password so it is not pre-populated on the form on subsequent uploads
+    this.setState(
+      { password: '' },
+      () => {
+        try {
+          const json: KeyringPair$Json | void = keyring.backupAccount(address, password);
+
+          if (!isUndefined(json)) {
+            const blob = new Blob([JSON.stringify(json)], { type: 'text/plain;charset=utf-8' });
+
+            FileSaver.saveAs(blob, `${address}.json`);
+
+            this.hidePasswordModal();
+          } else {
+            this.setState({ error: { key: t('error'), value: t('Unable to obtain account from memory') } });
+          }
+        } catch (e) {
+          this.setState({ error: { key: t('error'), value: t('Unable to save file') } });
+          console.error('Error retrieving account from local storage and saving account to file: ', e);
+        }
+      }
+    );
+  }
+
+  showPasswordModal = (): void => {
+    this.setState({ isPasswordModalOpen: true });
+  }
+
+  hidePasswordModal = (): void => {
+    this.setState({ isPasswordModalOpen: false });
+  }
+
+  render () {
+    const { address, isPasswordModalOpen } = this.state;
+    const { className, icon = 'download', isCircular = true, isPrimary = true, size = 'tiny', style } = this.props;
+
+    if (!address) {
+      return null;
+    }
+
+    return (
+      <div className={'accounts--Address-download'}>
+        { isPasswordModalOpen ? (
+            <Modal
+              dimmer='inverted'
+              open={isPasswordModalOpen}
+              onClose={this.hidePasswordModal}
+              size={'mini'}
+            >
+              <Modal.Content>
+                <div className='ui--grid'>
+                  <div className={'accounts--Address-modal'}>
+                    <AddressMini isShort={true} value={address} />
```

Comment:
Just `isShort` will do the trick here.

Is this not too small? Row or Summary maybe? (We can hide the balances/transactions if need-be))

Reply:
AddressMini is quite small, would be good for small screen sizes, i'll see how the other's look

Reply:
I just tried implementing both AddressSummary and AddressMini in both DownloadButton.tsx and UploadButton.tsx. I was going to use media queries to toggle one or the other depending on the device width in use, or even try something like the react-responsive library.

I implemented it as follows:
```
import AddressMini from '@polkadot/ui-app/AddressMini';
import AddressSummary from '@polkadot/ui-app/AddressSummary';
...
              <div className={'accounts--Address-modal'}>
                <AddressMini isShort value={address} />
                <AddressSummary value={address} />
```

But when I added the AddressSummary component it gave error `Uncaught ExtError: Application root must be wrapped inside 'rx-react/Api' to provide API context`.

So I added the following, thinking it would resolve the error: 
```
import withApi from '@polkadot/ui-react-rx/with/api';

...

const Component: React.ComponentType<any> = translate(
  withApi(DownloadButton)
);

export default Component;
```

But it didn't resolve the error. What do you think?